core/src/main/java/org/infinispan/stream/impl/CacheStreamIntermediateReducer.java

```diff
@@ -57,7 +59,8 @@ public void writeObject(ObjectOutput output, CacheStreamIntermediateReducer obje
 
       @Override
       public CacheStreamIntermediateReducer readObject(ObjectInput input) throws IOException, ClassNotFoundException {
-         return new CacheStreamIntermediateReducer((Queue) input.readObject(), (Function) input.readObject());
+         List<IntermediateOperation> intOps = (List<IntermediateOperation>) input.readObject();
+         return new CacheStreamIntermediateReducer(new ArrayDeque<>(intOps), (Function) input.readObject());
```

Comment:
You can change `CacheStreamIntermediateReducer` to take an Iterable instead and then you don't need this.